### PR TITLE
apps/testing: Add a cmocka framework for network testing

### PR DESCRIPTION
**Pull request by zs39**
## Summary
Provides a network automated testing framework, including the main directory structure and Makefile, CMakeLists, Kconfig and other files such as tcp.

## Impact
N/A

## Testing
qemu environment test

### CODE REVIEW

File: testing/nettest/Kconfig [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+config TESTING_NET_TEST_PRIORITY
+	int "Task priority"
+	default 100
+
+config TESTING_NET_TEST_STACKSIZE
+	int "Stack size"
+	default DEFAULT_TASK_STACKSIZE
+
+config TESTING_NET_TCP
```
Comment (GUIDINGLI) [inline testing/nettest/Kconfig]: what's the purpose of this marco ? keep only TESTING_NET_TEST ?

Comment (zs39) [reply testing/nettest/Kconfig]: > what's the purpose of this marco ? keep only TESTING_NET_TEST ?

In the future, test cases will be grouped according to network protocols. Should I remove this macro in this patch?

File: testing/nettest/tcp/test_tcp_connect_01.c [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: tcp_connect_setup_01
+ ****************************************************************************/
+
+int tcp_connect_setup_01(FAR void **state)
```
Comment (wengzhe) [inline testing/nettest/tcp/test_tcp_connect_01.c]: `test_tcp_connect_ipv4_setup` ?

File: testing/nettest/tcp/test_tcp_connect_01.c [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+{
+  return 0;
+}
+
+/****************************************************************************
+ * Name: test_tcp_connect_ipv4_01
+ ****************************************************************************/
+
+void test_tcp_connect_ipv4_01(FAR void **state)
```
Comment (wengzhe) [inline testing/nettest/tcp/test_tcp_connect_01.c]: What's the rule of our case name? `test_tcp_connect_ipv4` or `test_tcp_connect_ipv4_01` ?

File: testing/nettest/tcp/test_tcp_connect_01.c [hunk 4]
```diff
@@ -0,0 +1,46 @@
+/****************************************************************************
+ * apps/testing/nettest/tcp/test_tcp_connect_01.c
```
Comment (wengzhe) [inline testing/nettest/tcp/test_tcp_connect_01.c]: `test_tcp_connect_ipv4.c`?

File: testing/nettest/tcp/test_tcp_common.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: tcp_group_setup
+ ****************************************************************************/
+
+int tcp_group_setup(FAR void **state)
```
Comment (wengzhe) [inline testing/nettest/tcp/test_tcp_common.c]: `test_tcp_group_setup`

File: testing/nettest/tcp/test_tcp_common.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+{
+  return 0;
+}
+
+/****************************************************************************
+ * Name: tcp_group_teardown
+ ****************************************************************************/
+
+int tcp_group_teardown(FAR void **state)
```
Comment (wengzhe) [inline testing/nettest/tcp/test_tcp_common.c]: `test_tcp_group_teardown`

**Review by wengzhe — CHANGES_REQUESTED** [review]

**Pull request closed without merging**